src/sgp/SGP.cc

```diff
@@ -380,12 +380,21 @@ int main(int argc, char* argv[])
 
 	if(get_number_of_mods(params) > 0)
 	{
-		char* rustModName = get_mod(params, 0);
-		std::string modName = std::string(rustModName);
-		free_rust_string(rustModName);
-		std::string modResFolder = FileMan::joinPaths(FileMan::joinPaths(FileMan::joinPaths(extraDataDir, "mods"), modName), "data");
+		std::vector<std::string> modNames;
+		std::vector<std::string> modResFolders;
+		auto n = get_number_of_mods(params);
```

Comment:
This can be moved up, so it's looked up only once.

Reply:
Done.